.github/script-missing-mode-enforcement.py

```diff
@@ -0,0 +1,62 @@
+"""
+Script to scan the OpenTTD's script API for functions that miss checks for the
+function being called from the right mode (deity or company mode).
+
+When a function calls either ScriptObject::Command or ScriptObject::GetCompany
+then the function is considered dangerous. When one of the mode enforcement
+macros from script_error.hpp, i.e. EnforceDeityMode, EnforceCompanyModeValid or
+EnforceDeityOrCompanyModeValid, are called in the function, then we consider
+that the function has mode enforcement.
+
+Any dangerous function for which no enforcement is found are emitted as errors.
+"""
+
+import glob
+import re
+import sys
+
+def checkModeEnforcement(path):
+	errors = []
+	with open(path, 'r') as reader:
+		modeEnforcementFound = False
+		dangerousFunction = False
+		for line in reader:
+			# Line does not start with a tab and have <word>::<word>. That looks like the begin of a function, so reset the state.
+			if re.match("^[^\t].*\w::\w", line):
+				modeEnforcementFound = False
+				dangerousFunction = False
+				currentFunction = line
+			else:
+				if re.match("\t(EnforceDeityMode|EnforceCompanyModeValid|EnforceDeityOrCompanyModeValid|EnforceDeityOrCompanyModeValid_Void)\\(", line):
+					# Mode enforcement macro found
+					modeEnforcementFound = True
+
+				if re.match(".*(ScriptObject::Command|ScriptObject::GetCompany).*", line):
+					# Dangerous function found
+					dangerousFunction = True
+
+				# Line with only a closing bracket. That looks like the end of a function, so check for the dangerous function without mode enforcement
+				if re.match("^}$", line) and dangerousFunction and not modeEnforcementFound:
+					errors.append("%s: %s" % (path, currentFunction.rstrip('\n').replace("/* static */ ", "")))
+
+	return errors
+
+
+def main():
+	errors = []
+	for path in sorted(glob.glob("src/script/api/*.cpp")):
+		# Skip a number of files that yield only false positives
+		if not path.endswith(("script_object.cpp", "script_companymode.cpp", "script_controller.cpp", "script_game.cpp")):
+			errors = errors + checkModeEnforcement(path)
+
+	if errors:
+		print("Mode enforcement was expected in the following files/functions:")
+		for error in errors:
+			print(error)
+		sys.exit(1)
+	else:
+		print("OK")
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
These days we use a tool called `black` to reformat our Python code. It basically applies the default coding style of Python to a file (https://peps.python.org/pep-0008/ and others). This saves you a lot of time and annoyance.

`pip install black && black -l 120 <filename>` is what we run in all other Python projects we build for OpenTTD. It fixes a few of my comments.